include/pisa/wand_data.hpp

```diff
@@ -22,7 +23,7 @@ namespace pisa {
 
         template<typename LengthsIterator>
         wand_data(LengthsIterator len_it, uint64_t num_docs,
-                      binary_freq_collection const &coll, partition_type type = partition_type::fixed_blocks, const float lambda = 0.0f) {
+                      binary_freq_collection const &coll, partition_type type = partition_type::fixed_blocks, boost::variant<float, uint64_t> block_size = uint64_t(0)) {
```

Comment:
Correct my if I'm wrong, but I think now that we have a variant type, `partition_type` is redundant, right?

Reply:
This is correct! and it is such a great improvement :)

Reply:
Refactored in the latest.